Add TinyGo JSON/Regex serialization benchmarks

posborne opened the pull request:
- Create Dockerfile.tinygo template for TinyGo benchmarks
- Add tinygo-json benchmark with encoding/json
- Add tinygo-regex benchmark using standard library regexp

Sample size of one, but in some quick comparison the regex with tinygo is fairly slow taking ~32x which is somewhat surprising.  CC @dgryski to review to ensure the workloads here are useful in representing some proxy of real-world workloads.

Given how much slower tinygo appears to be here it may make sense to truncate the input file to reduce the per-iteration time of the regex benchmark.

---

As with the previous round of benchmarks, portions of this PR were generated using an LLM though reworked a fair bit.  I opted to use a single tinygo/ benchmarks subdirectory as it was easier to achieve reuse in that way that make it simpler to add other tinygo benchmarks as we see fit.

posborne committed:
Compile tinygo benches with -opt=2 -gc=leaking

There's still a large divide with the rust regex bench but this brings
things a bit closer from 32x slower to 19x slower in a quick benchmark.

fitzgen commented on benchmarks/tinygo/build.sh:
```diff
@@ -0,0 +1,34 @@
+    cd "$dir"
+
+    # Build with TinyGo
+    "$TINYGO" build -o "$OUTPUT_DIR/tinygo-$benchmark.wasm" -target=wasi -opt=2 -gc=leaking .
```
Can you add a comment here documenting why `-gc=leaking` is used here?

posborne committed:
Add comment on tinygo leaking gc selection

posborne committed:
Add TinyGo JSON/Regex serialization benchmarks

- Create Dockerfile.tinygo template for TinyGo benchmarks
- Add tinygo-json benchmark with encoding/json
- Add tinygo-regex benchmark using standard library regexp
